### PR TITLE
Refactored kubemark into cloud-provider independent code and GCE specific code

### PR DESCRIPTION
**Pull request by shyamjvs**
Ref issue #38967 

The following are the major changes as part of this refactoring:
- Moved cluster-kubemark/config-default.sh -> cluster-kubemark/gce/config-default.sh (as the config is gce-specific)
- Changed kubernetes/cluster/kubemark/util.sh to source the right scripts based on the cloud-provider
- Added test/kubemark/skeleton/util.sh which defines a well-commented interface that any cloud-provider should implement to run kubemark. (We have this interface defined only for gce currently)
  This includes functions like creating the master machine instance along with its resources, executing a given command on the master (like ssh), scp, deleting the master instance and its resources.
  All these functions have to be overrided by each cloud provider inside the file /test/kubemark/$CLOUD_PROVIDER/util.sh
- Added the file test/kubemark/cloud-provider-config.sh which sets the variable CLOUD_PROVIDER that is later picked up by various scripts (start-kubemark.sh, stop-kubemark.sh, run-e2e-tests.sh)
- Removed test/kubemark/common.sh and moved whatever provider-independent code it had into start-kubemark.sh (the only place where the scipt is called) and moved the little gce-specific code
  into test/kubemark/gce/util.sh.
- Finally, removed useless code and restructured start-kubemark.sh and stop-kubemark.sh scripts.

@kubernetes/sig-scalability-misc @wojtek-t @gmarek

### CODE REVIEW

File: test/kubemark/run-e2e-tests.sh [hunk 1]
```diff
@@ -14,16 +14,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export KUBERNETES_PROVIDER="kubemark"
+# TODO(shyamjvs): This variable should be read from a cloud_provider_config.sh file.
```
Comment (wojtek-t) [inline test/kubemark/run-e2e-tests.sh]: It's not possible to comment on the previous file in github, so commenting here.

The only gce-specific command in configure-kubectl.sh is where do we get kubectl from. The main part that is really configuring kubectl is provider-independent. So we should either:
- split this into two scripts (install kubectl and configure it)
- or assume that kubectl is in PATH and remove the first part.

I guess for now, the former is easier.

File: test/kubemark/run-e2e-tests.sh [hunk 2]
```diff
@@ -14,16 +14,18 @@
...
+export CLOUD_PROVIDER="${CLOUD_PROVIDER:-gce}"
+export KUBERNETES_PROVIDER="kubemark/${CLOUD_PROVIDER}"
```
Comment (wojtek-t) [inline test/kubemark/run-e2e-tests.sh]: Changing this one will break different things. Please avoid doing this.

Comment (wojtek-t) [inline test/kubemark/run-e2e-tests.sh]: I mean, a bunch of different things are relying on this env var.

File: test/kubemark/start-kubemark.sh [hunk 3]
```diff
@@ -80,6 +84,10 @@ done
 rm kubemark
 cd $CURR_DIR
 
+##################################################################################
```
Comment (wojtek-t) [inline test/kubemark/start-kubemark.sh]: nit: I know it's pretty common in different project, but in Kubernetes we are trying to avoid things like long lines with "#" "or" = or "-" but we are trying to avoid these here.

Comment (shyamjvs) [inline test/kubemark/start-kubemark.sh]: Removed them as I fixed the TODOs now.

File: test/kubemark/run-e2e-tests.sh (hunk 2 above)
Comment (shyamjvs) [inline test/kubemark/run-e2e-tests.sh]: Changed it back to the original. And instead incorporated this cloud-provider variance in cluster/kubemark/util.sh .

File: test/kubemark/run-e2e-tests.sh (hunk 1 above)
Comment (shyamjvs) [inline test/kubemark/run-e2e-tests.sh]: Took the second choice. Easier for now.

**Comment by shyamjvs** [comment]
@k8s-bot bazel test this

**Comment by rmmh** [comment]
@k8s-bot bazel test this

**Comment by shyamjvs** [comment]
Changed the PR description to clearly explain the changes as part of the refactoring.

**Comment by wojtek-t** [comment]
I'm generally fine with the high-level idea. But I will let @gmarek to more detailed review.

File: cluster/kubemark/util.sh [hunk 4]
```diff
@@ -15,5 +15,6 @@
 # limitations under the License.
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
-source ${KUBE_ROOT}/cluster/gce/util.sh
-source ${KUBE_ROOT}/cluster/kubemark/config-default.sh
+source ${KUBE_ROOT}/test/kubemark/cloud-provider-config.sh
+source ${KUBE_ROOT}/cluster/${CLOUD_PROVIDER}/util.sh
+source ${KUBE_ROOT}/cluster/kubemark/${CLOUD_PROVIDER}/config-default.sh
```
Comment (gmarek) [inline cluster/kubemark/util.sh]: Fix the strange red endline character:)

Comment (shyamjvs) [inline cluster/kubemark/util.sh]: Done.

File: test/kubemark/start-kubemark.sh [hunk 5]
```diff
@@ -14,15 +14,34 @@
...
+# Script that creates a Kubemark cluster for a given cloud provider.
 
-# Hack to make it work for OS X. Ugh...
 TMP_ROOT="$(dirname "${BASH_SOURCE}")/../.."
 KUBE_ROOT=$(readlink -e ${TMP_ROOT} 2> /dev/null || perl -MCwd -e 'print Cwd::abs_path shift' ${TMP_ROOT})
+KUBECTL="${KUBE_ROOT}/cluster/kubectl.sh"
+KUBEMARK_DIRECTORY="${KUBE_ROOT}/test/kubemark"
+RESOURCE_DIRECTORY="${KUBEMARK_DIRECTORY}/resources"
+
+source "${KUBE_ROOT}/test/kubemark/skeleton/util.sh"
```
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: You don't need to source that

File: test/kubemark/cloud-provider-config.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
```
Comment (gmarek) [inline test/kubemark/cloud-provider-config.sh]: 2017 :) - here and everywhere else.

File: test/kubemark/start-kubemark.sh [hunk 7]
```diff
@@ -14,15 +14,34 @@
...
+source "${KUBE_ROOT}/test/kubemark/cloud-provider-config.sh"
+source "${KUBE_ROOT}/test/kubemark/${CLOUD_PROVIDER}/util.sh"
+source "${KUBE_ROOT}/cluster/kubemark/${CLOUD_PROVIDER}/config-default.sh"
+
+# hack/lib/init.sh will ovewrite ETCD_VERSION if this is unset
+# to what is default in hack/lib/etcd.sh
+# To avoid it, if it is empty, we set it to 'avoid-overwrite' and
+# clean it after that.
+if [ -z "${ETCD_VERSION:-}" ]; then
```
Comment (gmarek) [inline test/kubemark/start-kubemark.sh]: ETCD_VERSION="${ETCD_VERSION:-avoid-overwrite}"

File: test/kubemark/start-kubemark.sh (hunk 5 above)
Comment (shyamjvs) [inline test/kubemark/start-kubemark.sh]: Well I followed it similar to kube-up. Importing the skeleton first and then the provider-specific util would make it easily visible to the developer that a particular function of the interface wasn't defined in the util.

File: test/kubemark/cloud-provider-config.sh (hunk 6 above)
Comment (shyamjvs) [inline test/kubemark/cloud-provider-config.sh]: Done. For all new files created or old files moved/renamed.

**Comment by shyamjvs** [comment]
@gmarek Fixed thew issues, PTAL now.
Also, I added better (more informative and colourful) logging statements in the start-kubemark script. Also made the interface for cloud-providers more compact by removing the function "wait-for-master-ssh-reachability" from the interface, as it can be implemented with already existing functions in the interface like "execute-cmd-on-master-with-retries". Fixed some bugs in the gce 'copy-files' interface function and in run-gcloud-compute-with-retries.

**Comment by shyamjvs** [comment]
@k8s-bot unit test this

File: test/kubemark/run-e2e-tests.sh [hunk 8]
```diff
@@ -39,6 +40,6 @@ else
 	ARGS=$@
 fi
 
-go run ./hack/e2e.go -v --check_version_skew=false --test --test_args="--e2e-verify-service-account=false --dump-logs-on-failure=false ${ARGS}"
+#go run ./hack/e2e.go -v --check_version_skew=false --test --test_args="--e2e-verify-service-account=false --dump-logs-on-failure=false ${ARGS}"
```
Comment (gmarek) [inline test/kubemark/run-e2e-tests.sh]: Revert this change.

**Comment by gmarek** [comment]
Reviewed 1 of 11 files at r1, 4 of 11 files at r2, 1 of 1 files at r3, 6 of 6 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[test/kubemark/start-kubemark.sh, line 25 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-KZSUXLkCsjqHEuyfVhS-r3-25:-Ka7RR5grDKZUAPyJV3n:b-gxztsp) ([raw file](https://github.com/kubernetes/kubernetes/blob/2ecb57cdfcaf1aecd43a8beace18a39df7c9dfa2/test/kubemark/start-kubemark.sh#L25)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Well I followed it similar to kube-up. Importing the skeleton first and then the provider-specific util would make it easily visible to the developer that a particular function of the interface wasn't defined in the util.
</blockquote></details>

Fair enough.

---

*[test/kubemark/start-kubemark.sh, line 34 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7RWV4MNY75XPixYVc:-Ka7RWV4MNY75XPixYVd:b47l1nj) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L34)):*
> ```Shell
> # To avoid it, if it is empty, we set it to 'avoid-overwrite' and
> # clean it after that.
> ETCD_VERSION="${ETCD_VERSION:-avoid-overwrite}"
> ```

I though about this change, but I deleted the suggestion eventually.  IIUC bash semantics the  `:-` substitutes value only if variable is undefined, while former `[  -z ... ]` will work also when the value is "". So please go back to previous version.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Ka7S9giwTc1kyNpHxaI:b-g2qqpw) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
> ```Shell
> # Wait until all hollow-nodes are running or there is a timeout.
> function wait-for-hollow-nodes-to-run-or-timeout {
>   echo -n "Waiting for all HollowNodes to become Running..."
> ```

In such case you may want to remove  `...` from the end of the line.

---

*[test/kubemark/start-kubemark.sh, line 290 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7SJaBXDR5f62PwGVt:-Ka7SJaBXDR5f62PwGVu:bs9pwd3) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L290)):*
> ```Shell
>     if [ $((now - start)) -gt 1800 ]; then
>       echo ""
>       echo -e "${color_red} Timeout waiting for all HollowNodes to become Running. ${color_norm}"
> ```

If you want to do this, verify if it's run from a terminal that supports coloring. Otherwise we'll have a lot of garbage in logs...

---

*[test/kubemark/gce/util.sh, line 30 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7TMypNnP21IyRZE2v:-Ka7TMypNnP21IyRZE2w:b-boo3uu) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/gce/util.sh#L30)):*
> ```Shell
>   for attempt in $(seq 1 ${RETRIES}); do
>     echo "" > /tmp/gcloud_retries_log
>     if ! gcloud compute "$@" |& tee /tmp/gcloud_retries_log; then
> ```

You still use the same file as a 'lock' for all calls - please differentiate the name in any way.

---

*[test/kubemark/gce/util.sh, line 101 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7TWuwqNzWIk80Elu5:-Ka7TWuwqNzWIk80Elu6:b-pvlov7) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/gce/util.sh#L101)):*
> ```Shell
> 
> function execute-cmd-on-master-with-retries() {
>   # No. of retries is '$2' (if provided) or 1 (default).
> ```

Move comment above the function and explicitly write what arguments it takes.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by shyamjvs** [comment]
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[test/kubemark/run-e2e-tests.sh, line 43 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-KZSUXLkCsjqHEuyfVhR-r4-43:-Ka7W9QJy87KMfvdWadf:b3605zy) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/run-e2e-tests.sh#L43)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Revert this change.
</blockquote></details>

Oops. Sorry for the blooper.

---

*[test/kubemark/start-kubemark.sh, line 25 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-KZSUXLkCsjqHEuyfVhS-r3-25:-Ka7Z-GCxjIKVhqjFD-x:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/2ecb57cdfcaf1aecd43a8beace18a39df7c9dfa2/test/kubemark/start-kubemark.sh#L25)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Fair enough.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark.sh, line 34 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7RWV4MNY75XPixYVc:-Ka7WFtcKfZpoJJpWz6T:bylci2w) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L34)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

I though about this change, but I deleted the suggestion eventually.  IIUC bash semantics the  `:-` substitutes value only if variable is undefined, while former `[  -z ... ]` will work also when the value is "". So please go back to previous version.
</blockquote></details>

Good catch. Fixed it.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Ka7WUbh2oFrg3mZk-K9:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

In such case you may want to remove  `...` from the end of the line.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark.sh, line 290 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7SJaBXDR5f62PwGVt:-Ka7WjRQc2wi3wuqeBBp:b-e1v651) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L290)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

If you want to do this, verify if it's run from a terminal that supports coloring. Otherwise we'll have a lot of garbage in logs...
</blockquote></details>

We had been doing it inside 'run-compute-with-retries' function all this while. So it seems to be fine on both jenkins and local machines. Besides, "color_red", "color_green", etc are bash variables that have been declared inside cluster/lib/util.sh and in the worst case, if these are not defined the echoes just happen in white color.

---

*[test/kubemark/gce/util.sh, line 30 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7TMypNnP21IyRZE2v:-Ka7atS0_SyLDMrW6yoc:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/gce/util.sh#L30)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

You still use the same file as a 'lock' for all calls - please differentiate the name in any way.
</blockquote></details>

Done.

---

*[test/kubemark/gce/util.sh, line 101 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7TWuwqNzWIk80Elu5:-Ka7YZ632uOwswSFZohw:b-klfsox) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/gce/util.sh#L101)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Move comment above the function and explicitly write what arguments it takes.
</blockquote></details>

I've put comments for each of the functions that are part of gce/util.sh inside skeleton/util.sh as that is where the interface is defined.
And the comment here was actually to explain the 'RETRIES="${2:-1}"' part. Do you suggest removing it?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by shyamjvs** [comment]
pcon.google.com

**Comment by shyamjvs** [comment]
@gmarek Fixed on comments, can you PTAL?

**Comment by gmarek** [comment]
Reviewed 3 of 4 files at r5.
Review status: 8 of 9 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Kaq3lJKyDW3ebSeUCU2:b8yts9f) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Space is also not needed:)

---

*[test/kubemark/gce/util.sh, line 101 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7TWuwqNzWIk80Elu5:-Kaq4iHJQntwJ1QMpxOg:b-ucex5b) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/gce/util.sh#L101)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

I've put comments for each of the functions that are part of gce/util.sh inside skeleton/util.sh as that is where the interface is defined.
And the comment here was actually to explain the 'RETRIES="${2:-1}"' part. Do you suggest removing it?
</blockquote></details>

This function takes arguments - for bash we list them above function definition.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by shyamjvs** [comment]
Review status: 8 of 9 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Kaqo1PgXiL4R9NJ0W7p:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Space is also not needed:)
</blockquote></details>

Done.

---

*[test/kubemark/gce/util.sh, line 101 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7TWuwqNzWIk80Elu5:-KaqoPunYF2bMO5kmxyy:b8dticn) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/gce/util.sh#L101)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

This function takes arguments - for bash we list them above function definition.
</blockquote></details>

Makes sense. Moved the comment above the function.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by gmarek** [comment]
Reviewed 1 of 4 files at r5, 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Kar7WotP8VM_neHwuWA:bnhhomg) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Neither space, nor ... should be here. Dots will be added by code that waits:)

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by shyamjvs** [comment]
Review status: 10 of 11 files reviewed at latest revision, 7 unresolved discussions.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Kar8h3jIajL59L8SU4u:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Neither space, nor ... should be here. Dots will be added by code that waits:)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by shyamjvs** [comment]
Review status: 10 of 11 files reviewed at latest revision, 7 unresolved discussions.

---

*[test/kubemark/start-kubemark.sh, line 278 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033#-Ka7S9giwTc1kyNpHxaH:-Kar8z6ynAXaOJr_F4yC:b-kah81q) ([raw file](https://github.com/kubernetes/kubernetes/blob/7e1a24efc0ffb354b1d473f62456fdb6446779b3/test/kubemark/start-kubemark.sh#L278)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I added those dots to ensure that we'd have at least a bare minimum of 3 dots (because we don't want to have 'Done' immediately after 'Waiting for hollownodes' string). Just added a space before 'Done' to get rid of this problem.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39033)*

**Comment by gmarek** [comment]
It seems that tests are stuck...

**Comment by spxtr** [comment]
@k8s-bot test this
https://github.com/kubernetes/test-infra/issues/1598

**Comment by shyamjvs** [comment]
@k8s-bot cross build this
